### PR TITLE
Makes paragraph spacing customizable.

### LINKED ISSUE #547: Review spacing of lists, paragraphs and blockquotes.
These elements should have more spacing after them than regular lines.  They have to be pleasing to read.

asd

### PR DESCRIPTION
**Pull request by diegoreymendez**
### Description:

Fixes #547 

Paragraph spacing is now customizable per-app.  It's also nicer by default.

### Screenies:

screen shot 2017-10-17 at 3 05 57 pm

### Testing:

**Test 1:**

Launch the editor with content and make sure it looks about right.

**Test 2:**

Launch an empty editor and start creating paragraphs, lists, block quotes.  Also use ENTER + SHIFT to create new lines in the same paragraph.

Make sure the spacing looks nicer than before and that it's consistent.

### Known issues:

Each list item is a different paragraph, so the spacing for lists is a bit off.  This will be handled in an upcoming PR.

### CODE REVIEW

File: Aztec/Classes/TextKit/LayoutManager.swift [hunk 1]
```diff
@@ -67,18 +67,20 @@ private extension LayoutManager {
                 let lineCharacters = textStorage.attributedSubstring(from: lineRange).string
                 let lineEndsParagraph = lineCharacters.isEndOfParagraph(before: lineCharacters.endIndex)
                 let blockquoteRect = self.blockquoteRect(origin: origin, lineRect: rect, blockquoteIndent: blockquoteIndent, lineEndsParagraph: lineEndsParagraph)
-
+
```
Comment (jleandroperez) [inline Aztec/Classes/TextKit/LayoutManager.swift]: Can we revert this one?

File: Aztec/Classes/TextKit/LayoutManager.swift [hunk 2]
```diff
@@ -252,12 +254,12 @@ private extension LayoutManager {
         let markerPlain = list.style.markerText(forItemNumber: number)
         let markerText = NSAttributedString(string: markerPlain, attributes: markerAttributes)
 
-        var yOffset = -style.paragraphSpacingBefore
-
-        if let font = markerAttributes[NSFontAttributeName] as? UIFont {
-            yOffset += (rect.height - font.lineHeight)
+        var yOffset = CGFloat(0)
+
```
Comment (jleandroperez) [inline Aztec/Classes/TextKit/LayoutManager.swift]: Can we nuke the extra spacing?

File: Aztec/Classes/TextKit/LayoutManager.swift [hunk 3]
```diff
@@ -252,12 +254,12 @@ private extension LayoutManager {
...
-
-        if let font = markerAttributes[NSFontAttributeName] as? UIFont {
-            yOffset += (rect.height - font.lineHeight)
+        var yOffset = CGFloat(0)
+        
+        if location > 0 {
+            yOffset += style.paragraphSpacingBefore
         }
-
+
```
Comment (jleandroperez) [inline Aztec/Classes/TextKit/LayoutManager.swift]: Same as above

**Comment by diegoreymendez** [comment]
@jleandroperez - Spaces removed.  Thanks!

File: Aztec/Classes/TextKit/ParagraphStyle.swift [hunk 4]
```diff
@@ -214,26 +188,42 @@ open class ParagraphStyle: NSMutableParagraphStyle, CustomReflectable {
...
             let textTab = NSTextTab(textAlignment: .natural, location: CGFloat(location), options: [:])
-
+            
             tabStops.append(textTab)
         }
-
+        
         style.tabStops = tabStops
-
+        style.lineSpacing = 8
```
Comment (SergioEstevao) [inline Aztec/Classes/TextKit/ParagraphStyle.swift]: Can this 8 be changed? It feels a bit like a magic number here.

File: Aztec/Classes/TextKit/ParagraphStyle.swift [hunk 5]
```diff
@@ -214,26 +188,42 @@ open class ParagraphStyle: NSMutableParagraphStyle, CustomReflectable {
...
+        style.paragraphSpacing = 8
+        style.paragraphSpacingBefore = 8
+        
         return style
+    }()
+    
+    static func setDefault(_ newDefault: ParagraphStyle) {
+        cachedDefault = newDefault
```
Comment (SergioEstevao) [inline Aztec/Classes/TextKit/ParagraphStyle.swift]: So we can now configure the default here?

Comment (diegoreymendez) [reply Aztec/Classes/TextKit/ParagraphStyle.swift]: That's the idea... this may change over time though, but that's the initial hook we will be using.

**Pull request merged**